horovod/spark/keras/remote.py

```diff
@@ -323,3 +329,10 @@ def fn(tf, keras):
         config.intra_op_parallelism_threads = 1
         keras.backend.set_session(tf.Session(config=config))
     return fn
+
+
+def _model_checkpoint_callback_fn():
```

Comment:
I think we can simplify this a bit:

1. Change it from accepting a function to just accepting something that adheres to the `ModelCheckpoint` API.
2. Use the `checkpoint_callback.filepath` property to figure out where to read the saved model from (note that this filepath might be a template string: https://www.tensorflow.org/tutorials/keras/save_and_load#checkpoint_callback_options).
3. If not specified, create the default `ModelCheckpoint(ckpt_file)`.

Reply:
I don't think that the filepath should come from the user because they have to be responsible to figure out the appropriate temp folder with correct permissions. It also increases the chances of errors if users specify a fixed path, and the checkpoints from previous runs are present on that path. If a user wants copies of the checkpoint, they can add extra ModelCheckpoint callbacks using the callbacks parameter. 

If we agree that the user should not pass the filepath arg, I am not sure how to avoid not returning a function here as we want the user to specify the parameters of ModelCheckpoint callback.

Reply:
There are many cases where the user will want to keep their checkpoints in a particular location.  Needing to provide two separate callbacks that do the same thing in different directories is a bad user experience.

If you want to support the user not having to specify the path (because they don't want to keep the checkpoint files around), then you can solve this by creating a subclass of `ModelCheckpoint` for this purpose:

```
KerasEstimator(checkpoint_callback=hvd.callbacks.ReturnedModelCheckpoint(
    monitor='val_acc',
    mode='max',
    save_best_only=True), ...)
```

Nice thing about this approach is (1) it's flexible if the user wants to provide an alternative location to write checkpoints, and (2) you can prevent them from setting unnecessary things like `filepath` and `save_weights_only`.